src/engraving/libmscore/automationvertex.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef __AUTOMATIONVERTEX_H__
+#define __AUTOMATIONVERTEX_H__
+
+#include "engravingitem.h"
+#include "mscore.h"
+
+namespace Ms {
+class AutomationTrack;
+
+class AutomationVertex : public EngravingItem
+{
+    double _value;
+    double _ticks;  //Number of whole notes offset from start of score
```

Comment:
how will this work if the user removes or inserts a time range (measure etc.) into the music - will all existing vertices be adjusted accordingly?
All ticks can just be integers no? (even _value might be better as a float or qreal, I haven't seen a lot of doubles being used so far)

Reply:
I was planning for the vertices to be stored in a list sorted by _ticks.  _ticks could be an integer or something else, but if it is an integer then we will need to determine what unit one tick is.  We could use the 1/480 of a whole note MS3 used - is that unit still relevant?

Reply:
Yeah I assumed that's what it was meant to be!
But what I meant is that if you had, say, vertices defined for ticks 480 to 4800 then the user removed the first (4/4) measure entirely, would that modify all extract vertices in all staffs to subtract 480 from the "tick" values?
What if the user removed measures at the end of the piece that meant the vertex tick value was 
now beyond the end of the piece? (I would think you couldn't just delete the vertex if it was defining a slope...)

Reply:
If a measure is deleted, I would presume any coresponding vertices would be deleted too and it would be up to the user to fix things if it caused a problem. Probably something to ask Martin about, though.